compiler/rustc_codegen_ssa/src/mir/block.rs

```diff
@@ -864,7 +865,15 @@ impl<'a, 'tcx, Bx: BuilderMethods<'a, 'tcx>> FunctionCx<'a, 'tcx, Bx> {
                         // promotes any complex rvalues to constants.
                         if i == 2 && intrinsic == sym::simd_shuffle {
                             if let mir::Operand::Constant(constant) = &arg.node {
-                                let (llval, ty) = self.simd_shuffle_indices(bx, constant);
+                                let (llval, ty) = self.early_evaluate_const_vector(bx, constant);
+                                let llval = llval.unwrap_or_else(|| {
+                                    bx.tcx().dcx().emit_err(errors::ShuffleIndicesEvaluation {
+                                        span: constant.span,
+                                    });
+                                    // We've errored, so we don't have to produce working code.
+                                    let llty = bx.backend_type(bx.layout_of(ty));
+                                    bx.const_undef(llty)
+                                });
```

Comment:
deduplicate this with the other use sites

Reply:
actually, just do this inside `early_evaluate_const_vector` like it was done before this PR and remove `ShuffleIndicesEvaluation`

Reply:
This should be taken care of in the following commit; a98491c4a51fb56c61027e96f305e096ad27a09f

I noticed that, 2 representations on the same struct should be handled in the same manner (as shown by the tests). The structs are below: 

```rs
#[repr(simd)]
#[derive(Clone)]
pub struct i8x2(i8, i8);

#[repr(simd)]
#[derive(Clone)]
pub struct i8x2_arr([i8; 2]);
```

I believe they should both be producing the same code (they do in the non-const case). However the `i8x2_arr` variant would ICE, thus the commit fixes that issue.